integrationtest/mapper/orm/src/test/java/org/hibernate/search/integrationtest/mapper/orm/search/SearchQueryBaseIT.java

```diff
@@ -474,6 +482,21 @@ public void select_searchProjection_multiple() {
 		} );
 	}
 
+	@Test
+	public void select_searchProjection_id_wrongIdentifierType() {
+		OrmUtils.withinSession( sessionFactory, session -> {
+			SearchSession searchSession = Search.session( session );
+
+			assertThatThrownBy( () -> searchSession.search( Book.class )
+					.select( f -> f.id( String.class ) ) // wrong identifier type
+			)
+					.isInstanceOf( SearchException.class )
+					.hasMessageContainingAll(
+							"The required identifier type 'java.lang.String' does not match the actual identifier type 'java.lang.Integer'"
+					);
+		} );
+	}
+
```

Comment:
I think you can remove this.

Invalid types are already tested on the backend side in `IdentifierSearchProjectionSingleValuedIT#invalidProjectionType`.

On the mapper side, I don't think you need a dedicated test for ORM; you're already dealing with this in `integrationtest/mapper/pojo-base`.